src/AuthorizationUtils.php

```diff
@@ -0,0 +1,103 @@
+use Symfony\Component\Mercure\Jwt\TokenFactoryInterface;
+use Symfony\Component\WebLink\GenericLinkProvider;
+use Symfony\Component\WebLink\Link;
+
+final class AuthorizationUtils
+{
+    private const MERCURE_AUTHORIZATION_COOKIE_NAME = 'mercureAuthorization';
+
+    private $hubs;
+    private $factories;
+
+    /**
+     * @param iterable<string, Hub>                   $hubs
+     * @param iterable<string, TokenFactoryInterface> $factories
+     */
+    public function __construct(iterable $hubs, iterable $factories)
+    {
+        $this->hubs = \is_array($hubs) ? $hubs : iterator_to_array($hubs);
+        $this->factories = \is_array($factories) ? $factories : iterator_to_array($factories);
+    }
+
+    /**
+     * Add mercure link header to the given request.
+     */
+    public function addLink(string $hub, Request $request): void
+    {
+        // Prevent issues with NelmioCorsBundle
+        if ($this->isPreflightRequest($request)) {
+            return;
+        }
+
+        if (!isset($this->hubs[$hub])) {
+            throw new InvalidArgumentException(sprintf('Invalid hub name "%s", expected one of "%s".', $hub, implode('", "', array_keys($this->hubs))));
+        }
+
+        $url = $this->hubs[$hub]->getPublicUrl();
+        $link = new Link('mercure', $url);
+
+        if (null === $linkProvider = $request->attributes->get('_links')) {
+            $request->attributes->set('_links', new GenericLinkProvider([$link]));
+
+            return;
+        }
+
+        $request->attributes->set('_links', $linkProvider->withLink($link));
+    }
+
+    /**
+     * Create Authorization cookie for the given hub.
+     *
+     * @param string   $hub              the hub to generate the cookie for
+     * @param string[] $subscribe        a list of topics that the authorization cookie will allow subscribing to
+     * @param string[] $publish          a list of topics that the authorization cookie will allow publishing to
+     * @param mixed[]  $additionalClaims an array of additional claims for the JWT
+     *
+     * @return Cookie The
+     */
+    public function createCookie(string $hub, array $subscribe = [], array $publish = [], array $additionalClaims = []): Cookie
+    {
+        if (!isset($this->hubs[$hub])) {
+            throw new InvalidArgumentException(sprintf('Invalid hub name "%s", expected one of "%s".', $hub, implode('", "', array_keys($this->hubs))));
+        }
+
+        if (!isset($this->factories[$hub])) {
+            throw new InvalidArgumentException(sprintf('A token factory is not defined for hub "%s".', $hub));
+        }
+
+        $token = $this->factories[$hub]->create($subscribe, $publish, $additionalClaims);
+
+        $url = $this->hubs[$hub]->getPublicUrl();
+        /** @var array $urlComponents */
+        $urlComponents = parse_url($url);
+
+        return Cookie::create(self::MERCURE_AUTHORIZATION_COOKIE_NAME)
+            ->withValue($token)
+            ->withDomain($urlComponents['host'])
```

Comment:
For increased security, this should be set only if the domain of the hub is a subdomain of the current domain. If it's the same domain, it's better to not set this flag, and if they don't share a common top domain, we must throw because the cookie will not be sent to the Mercure hub anyway.

Also, the `secure` flag should be set by default (at least if the URL of the hub uses the `https` scheme), the `httponly` flag must be set and the `samesite` attribute must be set to `strict`.

Reply:
done 👍